src/fidesops/schemas/saas/saas_config.py

```diff
@@ -186,6 +202,7 @@ class SaaSConfig(BaseModel):
 
     fides_key: FidesOpsKey
     name: str
+    type: SaaSType
```

Comment:
Wanted to highlight that if we have any users that already have SaasConfigs saved, they will not have this required `type` field, and they will get validation errors any place this `SaasConfig` is used until they update their saas config.

If existing users are a concern, we could add a migration that updated any existing saas configs with a type (perhaps seeing if a known saas type is in the name or fides_key, otherwise using "custom".

Reply:
Good idea! I'll give this a shot

Reply:
hey @pattisdr I'm struggling to find a good way to extract the existing `type`s from the `SaasConfig`. The `SaasConfig` field is a `JSONB` column, so we'll need to do some data manipulation within alembic. Do you know of another scenario where we've done something similar?

Reply:
Since I'm not worried about performance, I would suggest looping through the handful of existing saas configs, updating the type if we think its `fides_key` or `name` overlaps with a SaaSType, otherwise, just use "custom". You have to be careful about importing sqlalchemy models into alembic migrations so I'd use textclauses instead.   The most similar example is the [default policies migration](https://github.com/ethyca/fidesops/blob/main/src/fidesops/migrations/versions/55d61eb8ed12_add_default_policies.py) I added. 


This isn't a complete solution, no error handling or data types added, you might look at the name too, but something like this inside an alembic migration:

```python
from alembic import op
from sqlalchemy import text
from fidesops.schemas.saas.saas_config import SaaSType


connection = op.get_bind()
query = text("""select datasetconfig.id, dataset from datasetconfig, connectionconfig where connectionconfig.id = datasetconfig.connection_config_id and connectionconfig.connection_type = 'saas';
""")
saas_options = [saas_type.value for saas_type in SaaSType]
update_query = text("""update datasetconfig set dataset = jsonb_set(dataset, '{type}', :saas_type)""") 

for id, dataset in connection.execute(query):
    fides_key = dataset["fides_key"]
    saas_type = next((opt for opt in saas_options if fides_key.lower().startswith(opt)), "custom")
    connection.execute(update_query, {"saas_type": f"\"{saas_type}\""})
   
```

Reply:
Thanks so much for writing this up @pattisdr ! As far as alembic downgrade, should we simply use "custom"? Since the field can't be `None`.

Reply:
I think that's a good idea

Reply:
for posterity, my example above is incorrectly updating the datasetconfig where we want to be updating connectionconfig > saas_config.